modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java

```diff
@@ -52,47 +55,128 @@
 public class PingManager {
 
     /**
-     * Collects new URLs reported by {@link PingManager#inventory} and synchronizes the various threads reporting the
-     * new URLs and those ones consuming them.
+     * An an add or remove operation that can be performed on a {@link Collection}.
      *
      * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
      */
-    static class NewUrlsCollector implements Action1<Resource> {
-        private final Object lock = new Object();
-        private List<PingDestination> newUrls = new ArrayList<>();
+    static class UrlChange {
+
+        /** The nature of the change - {@link Action#created()} or {@link Action#deleted()} */
+        private final Action<?, ?> action;
+
+        /** The {@link PingDestination} that the change is related to */
+        private final PingDestination destination;
 
         /**
-         * A callback for the {@link Inventory} that collects newly added URLs. It is safe to call this method
-         * concurrently from any random thread.
+         * Creates a new {@link UrlChange}.
          *
-         * @see rx.functions.Action1#call(java.lang.Object)
+         * @param action the nature of the change - {@link Action#created()} or {@link Action#deleted()}
+         * @param destination the {@link PingDestination} that the change is related to
          */
-        @Override
-        public void call(Resource r) {
-            if (PingDestination.isUrl(r)) {
-                synchronized (lock) {
-                    newUrls.add(PingDestination.from(r));
-                }
+        public UrlChange(Action<?, ?> action, PingDestination destination) {
+            super();
+            this.action = action;
+            this.destination = destination;
+        }
+
+        /**
+         * Applies this {@link UrlChange} to the given {@link Collection} of {@link PingDestination}s - i.e.
+         * {@link #destination} is either added or removed from the collection.
+         *
+         * @param destinations the {@link Collection} of {@link PingDestination}s that this {@link UrlChange} should be
+         *        applied to.
+         */
+        public void apply(Collection<PingDestination> destinations) {
+            if (Action.created().equals(this.action)) {
+                destinations.add(this.destination);
+            } else if (Action.deleted().equals(this.action)) {
+                destinations.remove(this.destination);
+            } else {
+                throw new IllegalStateException("Unexpected action '" + action
+                        + "'; expected Action.created() or Action.deleted()");
             }
         }
+    }
 
+    /**
+     * Collects URL additions and removals reported by {@link PingManager#inventory} and synchronizes the various
+     * threads reporting the new URLs and those ones consuming them.
+     *
+     * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+     */
+    static class UrlChangesCollector {
```

Comment:
Should that go into its own class file to not "pollute" this manager class? Probably along with the `UrlChange`